include/ada/url_search_params-inl.h

```diff
@@ -161,10 +161,95 @@ inline void url_search_params::remove(const std::string_view key,
   });
 }
 
+// returns -1 if str1 < str2, 0 if str1 == str2, 1 if str1 > str2
+// the comparison is done using the UTF-16 code units of the UTF-8 code points
+inline int compare_utf8_as_utf16(std::string_view str1, std::string_view str2) {
```

Comment:
If we are not adding this ti url_search_params.h, I think we should surround this function with an anonymous namespace

Reply:
Also can this be constexpr?

Reply:
I have turned it into a lambda.